pkg/collector/corechecks/checkbase.go

```diff
@@ -114,14 +116,27 @@ func (c *CheckBase) CommonConfigure(integrationConfigDigest uint64, initConfig,
 			s.DisableDefaultHostname(true)
 		}
 
+		var customTags []string
+
+		// Set tags configured in DD_TAGS
+		ddtags := config.GetConfiguredTags(config.Datadog, false)
+		if len(ddtags) > 0 {
+			customTags = ddtags
+		}
+
 		// Set custom tags configured for this check
 		if len(commonOptions.Tags) > 0 {
+			customTags = append(customTags, commonOptions.Tags...)
+		}
+
+		// Set custom tags to sender
+		if len(customTags) > 0 {
```

Comment:
How will this affect checks that run on regular, non-cluster agents?

Reply:
Shouldn't affect as node agent already has it as host level tag, I think. Let me know if I'm missing.

Reply:
Right, this would not introduce new tags to checks data in the app, but it would still send more (eventually redundant) tags with every metric and event. Some potential problems with this:

- Higher memory usage by the metrics aggregator caused by more tags on each metric.
- Higher egress traffic from the agent.
- If a check metric already has a lot of tags, additional tags added here may push it over the intake limit.

WDYT?

Reply:
The aggregator sorts the tags array and removes duplicates for every metric sample. I did some benchmarks to evaluate the gain from doing this for static tags earlier in the pipeline and it showed negligible gain.